### ISSUE TITLE
Trade chat improvements

### ISSUE THREAD

**Issue by AnonymousII** [HasKha/GWToolboxpp#115]
as we all know when you stand in Kamadan trying to read trade chat its sometimes really hard to keep track of it since there are so many messages in a really short time - thus some trade chat enhancements would make this a lot easier:
- highlight certain keywords (or possibly even the whole message containing this keyword) in a different color (like red), where the keywords can be defined in toolbox settings
- filter all trade chat messaged that DONT contain a certain keyword which can be defined in the toolbox settings
- flash gw taskbar icon when a certain keyword appears in trade chat
- (possibly add a kind of chat request monitor on how many chat messages you can still write within the next interval before they would get rejected - there's a certain limit like max. 5 trade chat messages per minute before they get rejected and you have to wait a certain time until you can write trade chat messages again.)

At least the first 3 things should be easy to implement since toolbox itself can already write colorized messages and its basically just an inverse chat filter. Nevertheless these features would make it a lot easier to keep track of the trade chat.

Number four would just be a nice to have but should also be easy to implement since you just have to count the trade chat messages a character sent within the last time interval.

**Comment by ritlew** [comment]
I'm not sure the fourth bullet is straight forward, because the chat suppression is probably handled on server. This would mean that we would have to basically figure out how the server determines whether you can send another message or not. That's just my intuition.

The rest sounds straight forward.

**Comment by AnonymousII** [comment]
well i guess its just a fixed pattern like "max. 5 messages in 1 minute, 2 minutes cooldown, 5 messages,..." (but it's just guessed). Since they probably won't change it within near future it may be enough to try it out once and hard code this pattern - even if it gets changed it should be easy to update once you got all the stuff around it like the widget etc.

**Comment by HasKha** [comment]
Thanks for the suggestions.
Those things are possible. My main problem is that I dislike the trade chat system, and I would rather work to make something much better than to slightly improve the existing system. This would mean creating a proper auction-house system, which is a hard, but possible, long-term project.

**Comment by AnonymousII** [comment]
Yeah no doubt - a proper auction-house system would be much better than any of these trade chat enhancements but it will definitely be a lot of work... Nevertheless these enhancements would be a good intermediate step while the auction house is under construction and shouldn't take more that a few hours to implement since similair functionality is already implemented (at least for 1-3) :)
